nixos/modules/services/security/proxychains.nix

```diff
@@ -0,0 +1,184 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+
+  cfg = config.services.proxychains;
+
+  configFile = ''
+    ${cfg.chain.type}
+    ${optionalString (cfg.chain.type == "random")
+    "chain_len = ${builtins.toString cfg.chain.length}"}
+    ${optionalString cfg.proxyDNS "proxy_dns"}
+    ${optionalString cfg.quietMode "quiet_mode"}
+    ${cfg.remoteDNSSubnet}
+    ${cfg.tcpReadTimeOut}
+    ${cfg.tcpConnectTimeOut}
+    ${cfg.localnet}
+    ${cfg.proxies}
+    ${cfg.extraConfig}
+  '';
+
+  proxyOptions = {
+    options = {
+      type = mkOption {
+        type = types.enum [ "http" "socks4" "socks5" ];
+        description = "Proxy type.";
+      };
+
+      host = mkOption {
+        type = types.str;
+        description = "Proxy host or IP address.";
+      };
+
+      port = mkOption {
+        type = types.port;
+        description = "Proxy port";
+        apply = builtins.toString;
```

Comment:
Don't use `apply`, it prevents other modules from reusing module options values. Instead use `toString` where `port` is used. Same for all the other uses of `apply`.